### PR TITLE
jakub_xlsx to master pt.2

### PR DESCRIPTION
**Pull request by ZaiZu2**
1. Corrected id column removal in update_from_df(), corrected json data type casting
2. Added requirements_dev.txt and requirements_prod.txt, adjusted .gitignore to allow these files

### CODE REVIEW

File: dbhydra/dbhydra_core.py [hunk 1]
```diff
@@ -291,6 +290,7 @@ def __init__(self, config_file="config.ini", db_details=None):
 
         self.lock = threading.Lock()
 
+        # NOTE: Pointless call (?): This is simply returning ContextManager object as `self.connect_to_db()` is a contextmanager decorated
```
Comment (DovaX) [inline dbhydra/dbhydra_core.py:293]: It is not pointless call - it is testing of connection

File: dbhydra/dbhydra_core.py [hunk 2]
```diff
@@ -481,17 +485,17 @@ class Mysqldb(AbstractDB):
     'Jsonable': "json"
     }
 
-    def connect_to_db(self):
+    def connect_to_db_globally(self):
+        """Connect to the DB globally (not as a ContextManager)"""
```
Comment (DovaX) [inline dbhydra/dbhydra_core.py:489]: what about backward compatibility? Breaking change here

File: dbhydra/dbhydra_core.py [hunk 3]
```diff
@@ -467,6 +467,10 @@ class Jsonable(str):
 
 
 class Mysqldb(AbstractDB):
+    def __init__(self, config_file="config.ini", db_details=None, global_: bool = False, *args, **kwargs):
+        super().__init__(config_file, db_details)
+        if global_:
```
Comment (DovaX) [inline dbhydra/dbhydra_core.py:472]: this naming is ugly - do you have a better name?

Comment (DovaX) [reply dbhydra/dbhydra_core.py:472]: moreover - adding parameters to the initialization of db should have a really key reason

File: requirements_prod.txt
Comment (DovaX) [inline requirements_prod.txt:1]: Arent these requirements actually forloop requirements? Dbhydra doesnt have that many requirements

**Review by DovaX — CHANGES_REQUESTED** [review]

File: dbhydra/dbhydra_core.py (hunk 1 above)
Comment (ZaiZu2) [reply dbhydra/dbhydra_core.py:293]: How is this testing connection? When connection is tested on platform, context manager `connect_to_db` method is used.  Here it's simply returning ContextManager object, which does not even get consumed.

File: dbhydra/dbhydra_core.py (hunk 2 above)
Comment (ZaiZu2) [reply dbhydra/dbhydra_core.py:489]: It's backward-compatible, `connect_to_db` is provided by `AbstractDB` class

File: requirements_prod.txt
Comment (ZaiZu2) [reply requirements_prod.txt:1]: These requirements were generated only for DBHydra, I did not include any platform/forloop requirements.

**Review by DovaX — CHANGES_REQUESTED** [review]
I accepted only part of the changes - and I commited them in the main branch

**Pull request closed without merging**